src/traces/bar/style.js

```diff
@@ -20,8 +20,46 @@ var attributeInsideTextFont = attributes.insidetextfont;
 var attributeOutsideTextFont = attributes.outsidetextfont;
 var helpers = require('./helpers');
 
+function resizeText(gd, gTrace, traceType) {
+    var fullLayout = gd._fullLayout;
+    var minSize = fullLayout['_' + traceType + 'Text_minsize'];
+    if(minSize) {
+        var shouldHide = fullLayout.uniformtext.mode === 'hide';
+
+        var t;
+        switch(traceType) {
+            case 'funnelarea' :
+            case 'pie' :
+            case 'sunburst' :
+            case 'treemap' :
+                t = gTrace.selectAll('g.slicetext').selectAll('text');
+                break;
+            default :
+                t = gTrace.selectAll('g.points').selectAll('g.point').selectAll('text');
+        }
+
+        var isCenter = (
+            traceType === 'pie' ||
+            traceType === 'sunburst'
+        );
+
+        t.each(function(d) {
+            var transform = d.transform;
```

Comment:
@etpinard here we read transform data from the calc data.

Reply:
Right ok I see - but it would be cleaner to read in the calc data from their parent `<g>`. Propagating d3-bound data using `selectAll` calls can lead to weird problems, for example:

https://github.com/plotly/plotly.js/blob/b2102a504a480493d9e474e15489c4874d5f6bac/src/traces/scattercarpet/plot.js#L32-L34

----

So here, it would be best to have your `t` variable select the `g.points` or `g.slicetext` and here have:

```js
t.each(function(d) {
  var el = d3.select(this).select('text')

  // and then apply changes to `el`
});

Reply:
Good call. Addressed in https://github.com/plotly/plotly.js/pull/4420/commits/eab205ae31606fa42f5e057f19da27d3389ca3f1.

Reply:
Very nice. Thanks! Can you now replace back the new `.data([pt])` with old `.data([0])` as pointed out in https://github.com/plotly/plotly.js/pull/4420#pullrequestreview-331517397

Reply:
In order to access transform (for `uniformtext`) similar to other traces, this is still needed for `pie` and `funnelarea`.

Reply:
Can't you grab the `transform` object from the parent `<g>`?

Reply:
Fixed by https://github.com/plotly/plotly.js/pull/4420/commits/603e0cce4ada75443932ab415e7b0f193e60c9ae thanks to @etpinard.

Reply:
Resolved!